File: front-end/highschool-ms/src/app/models/Classroom.ts
import Grades from "./Grades"
import { User } from "./User"

export interface Classroom{
    id: number,
    name: string,
    grade: Grades,
    classTeacher: User,
    maximumOccupants: number,
    registeredStudents: number,
    numberOfSubjecteds: number,
    academicYear: number,
    roomNumber: string
    learners: User[],
    [key: string]: any
}

export interface UserClassesModel{
    usersId: number,
    class: Classroom
}

File: back-end/HighSchoolManagementApi/Models/Classroom.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace HighSchoolManagementApi.Models
{
    [Table("Classrooms")]
    public class Classroom
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int? GradeId { get; set; }
        public Grades? Grade { get; set; } // Navigation property
        public int? ClassTeacherId { get; set; } = null;
        public Users? ClassTeacher { get; set; } = null;// Navigation property
        public int MaximumOccupants { get; set; }
        public int RegisteredStudents { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public List<Users> Learners {get; set;} = new List<Users>();
    }
}